src/ngram_matches.cc

```diff
@@ -69,23 +55,20 @@ namespace fuzzy
 
       // Get or create the AgendaItem corresponding to the sentence (of the suffix that matched)
       const auto sentence_id = _suffixArray.suffixid2sentenceid()[i].sentence_id;
-      auto* agendaItem = get_agendaitem(sentence_id);
-      if (!agendaItem)
-        agendaItem = new_agendaitem(sentence_id, _p_length);
+      auto& agendaItem = _psentences.emplace(sentence_id,
+                                             AgendaItem(sentence_id, _p_length)).first.value();
```

Comment:
The behavior may have changed here.
Is overwriting an existing `AgendaItem` impossible ?
Or it is not a problem ?

Reply:
Also are you sure that this avoids a memory allocation in the `if (agendaItem)` case ?

Reply:
If the key already exists, `emplace` returns an iterator on the existing value and does not insert the given `AgendaItem`. In this case, it is true that the `AgendaItem` is still constructed. I updated the code with `try_emplace` to avoid the construction in case the key already exists.